aixplain/model_interfaces/interfaces/function_models.py

```diff
@@ -241,6 +241,32 @@ def predict(self, request: Dict[str, str], headers: Dict[str, str] = None) -> Di
     
     def tokenize(self, messages: List[TextInput]) -> List[int]:
         pass
+    
+class TextGenerationChatModel(TextGenerationModel):
+    def run_model(self, api_input: Dict[str, List[TextInput]], headers: Dict[str, str] = None) -> Dict[str, List[TextGenerationOutput]]:
+        pass
+
+    def predict(self, request: Dict[str, str], headers: Dict[str, str] = None) -> Dict:
```

Comment:
we should also handle routing of requests here, like we do in `TextGenerationModel`. Although I don't like how it's done there, you are routing based on the existence of messages field.
I was thinking of client(backend in our case) passing in an explicit flag that tells us what to call. Can you take a look into how I implemented it in the [loglikelihood PR](https://github.com/aixplain/model-hosting/pull/194/files#diff-7c592bc79e86d480e4d82d7574c1bedda10f5e390b3ad95aea65a2171d39c2b2), we can discuss this in a meeting